### PR TITLE
Update Remote Services Docs with Recent Cloud SDK Improvements

### PR DESCRIPTION
**Pull request by MatKuhr**
If the `sap-client` destination property is defined it will always take effect, so probably not worth documenting here explicitly again..

Also, update the sample code, as `ClientCredentialsHttpDestination` is removed with SDK v5. The current code examples assume SDK v4 (from 4.16). For SDK 5, users would need to replace:

```diff
- .withProperties(Collections.singletonMap("name", "my-destination"))
+ .name("my-destination")
```

### CODE REVIEW

File: java/remote-services.md [hunk 1]
```diff
@@ -279,36 +275,27 @@ DefaultHttpDestination
...
-        clientCredentials,
-        URI.create("https://xsuaa.url"),
-        new XsuaaService()
-    );
+
+OAuth2DestinationBuilder
+        .forTargetUrl("https://example.org")
+        .withTokenEndpoint("https://xsuaa.url")
+        .withClient(clientCredentials, OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT)
+        .name("my-destination")
```
Comment (MatKuhr) [inline java/remote-services.md:285]: for SDK 4 we would need:

```suggestion
        .withProperties(Collections.singletonMap("name", "my-destination"))
```

File: java/remote-services.md [hunk 2]
```diff
@@ -279,36 +275,27 @@ DefaultHttpDestination
...
-        ClientCredentialsGrantType.USER_TOKEN,
-        URI.create("https://xsuaa.url"),
-        new XsuaaService()
-    );
+
+OAuth2DestinationBuilder
+        .forTargetUrl("https://example.org")
+        .withTokenEndpoint("https://xsuaa.url")
+        .withClient(clientCredentials, OnBehalfOf.NAMED_USER_CURRENT_TENANT)
+        .name("my-destination")
```
Comment (MatKuhr) [inline java/remote-services.md:299]: for SDK 4 we would need:

```suggestion
        .withProperties(Collections.singletonMap("name", "my-destination"))
```

File: java/remote-services.md (hunk 1 above)
Comment (beckermarc) [reply java/remote-services.md:285]: Could leverage new JDK methods: `.withProperties(Map.of("name", "my-destination"))`

File: java/remote-services.md [hunk 3]
```diff
@@ -279,36 +275,27 @@ DefaultHttpDestination
...
-        ClientCredentialsGrantType.USER_TOKEN,
-        URI.create("https://xsuaa.url"),
-        new XsuaaService()
-    );
+
+OAuth2DestinationBuilder
+        .forTargetUrl("https://example.org")
+        .withTokenEndpoint("https://xsuaa.url")
+        .withClient(clientCredentials, OnBehalfOf.NAMED_USER_CURRENT_TENANT)
+        .withProperties(Collections.singletonMap("name", "my-destination"))
```
Comment (MatKuhr) [inline java/remote-services.md:299]:
```suggestion
        .withProperties(Map.of("name", "my-destination"))
```

**Review by beckermarc — APPROVED** [review]

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**